src/context.rs

```diff
@@ -467,6 +416,60 @@ impl<A: Actor> Context<A> {
     }
 }
 
+pub struct Inbox<A> {
+    inner: BoxStream<'static, Either<BroadcastMessage<A>, AddressMessage<A>>>,
+}
+
+impl<A> Inbox<A>
+where
+    A: 'static,
+{
+    fn empty() -> Self {
+        Self {
+            inner: futures_util::stream::empty().boxed(),
+        }
+    }
+
+    fn new(
+        address_receiver: Receiver<AddressMessage<A>>,
+        broadcast_receiver: barrage::SharedReceiver<BroadcastMessage<A>>,
+    ) -> Self {
+        let broadcast_stream =
+            futures_util::stream::unfold(broadcast_receiver, |broadcast_receiver| async move {
+                match broadcast_receiver.recv_async().await {
+                    Ok(msg) => Some((msg, broadcast_receiver)),
+                    Err(_) => None,
+                }
+            })
+            .map(Either::Left)
+            .boxed();
+
+        let address_stream =
+            futures_util::stream::unfold(address_receiver, |address_receiver| async move {
+                match address_receiver.recv_async().await {
+                    Ok(msg) => Some((msg, address_receiver)),
+                    Err(_) => None,
+                }
+            })
+            .map(Either::Right)
+            .boxed();
+
+        Self {
+            inner: futures_util::stream_select!(broadcast_stream, address_stream).boxed(),
```

Comment:
The fact that stream_select is pseudorandom could cause potential issues with broadcast shutdown messages - if there is an external address constantly sending messages, the broadcast stream could theoretically be starved.